libkbfs/folder_branch_ops.go

```diff
@@ -886,6 +1119,26 @@ func (fbo *folderBranchOps) commitFlushedMD(
 
 		fbo.log.CDebugf(ctx, "Prefetch for revision %d complete; commiting",
 			rev)
+	case keybase1.FolderSyncMode_PARTIAL:
+		// For partially-synced TLFs, wait for the partial sync to
+		// complete, or for an update to happen.
+		lState := makeFBOLockState()
+		fbo.kickOffPartialSyncIfNeeded(ctx, lState, rmd)
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		go func() {
+			select {
+			case <-updatedCh:
+				cancel()
+			case <-fbo.shutdownChan:
+				cancel()
+			case <-ctx.Done():
+			}
+		}()
+		err := fbo.partialSyncs.Wait(ctx)
```

Comment:
So we don't return from this commit until the partial sync completes?

Reply:
Not necessarily, because the `ctx` can be canceled in a number of cases above, such as when the latest merged version is updated.  But yeah in the absence of any other revisions or a shut down, we wait for the partial sync to complete before commiting the MD to disk.  (It also gets committed to disk if the MD is accessed for reading.)